File: RentalAttireBackend/RentalAttireBackend/Application/Rentals/DTOs/RentalPageResponse.cs
using RentalAttireBackend.Domain.Entities;

namespace RentalAttireBackend.Application.Rentals.DTOs
{
    public class RentalPageResponse
    {
        public List<RentalDTO> Items { get; set; } = new();
        public int CurrentPage { get; set; }
        public int ItemsPerPage { get; set; }
        public int TotalCount { get; set; }
        public int TotalPages => (int)Math.Ceiling(TotalCount / (double)ItemsPerPage);
        public double TotalRevenue { get; set; }

        public Dictionary<string, int> StatusCounts { get; set; } = new();
        public int OverdueCount { get; set; }
        public int DueSoonCout { get; set; }
    }
}

File: RentalAttireFrontend/src/app/data/models/DTOs/Rentals/rental-page-response.ts
import { RentalDTO } from './rental';
import { RentalStatus } from './rental-status';

export interface RentalPageResponse {
  items: RentalDTO[];
  currentPage: number;
  itemsPerPage: number;
  totalCount: number;
  totalRevenue: number;
  // Computed over all active rentals, independent of the current search/filter/page
  statusCounts: Partial<Record<RentalStatus, number>>;
  overdueCount: number;
  dueSoonCount: number;
}